sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -114,10 +114,14 @@ object ReorderAssociativeOperator extends Rule[LogicalPlan] {
  * 1. Removes literal repetitions.
  * 2. Replaces [[In (value, seq[Literal])]] with optimized version
  *    [[InSet (value, HashSet[Literal])]] which is much faster.
+ * 3. Replaces [[In (value, Seq.empty)]] with false literal.
  */
 case class OptimizeIn(conf: CatalystConf) extends Rule[LogicalPlan] {
   def apply(plan: LogicalPlan): LogicalPlan = plan transform {
     case q: LogicalPlan => q transformExpressionsDown {
+      case expr @ In(v, list) if list.isEmpty =>
```

Comment:
How about we add a new case to handle null literal in value? Like the following:
```
      case expr @ In(v @ Literal(null, _), list) =>
        v
        
      case expr @ In(v, list) if list.isEmpty =>
        FalseLiteral
```